src/main/java/it/pagopa/pn/deliverypush/middleware/queue/consumer/PnEventInboundService.java

```diff
@@ -42,9 +42,12 @@ public PnEventInboundService(EventHandler eventHandler, PnDeliveryPushConfigs cf
 
     @Bean
     public MessageRoutingCallback customRouter() {
-        return message -> {
-            setTraceId(message);
-            return handleMessage(message);
+        return new MessageRoutingCallback() {
+            @Override
+            public FunctionRoutingResult routingResult(Message<?> message) {
+                setTraceId(message);
+                return new FunctionRoutingResult(handleMessage(message));
+            }
```

Comment:
???

Reply:
Nelle nuove versioni di Spring Cloud, MessageRoutingCallback non è più una functional interface, per cui la notazione lambda non è più possibile (il metodo che veniva utilizzato per effettuare la lambda è anche deprecato adesso):

 /** @deprecated */
    @Deprecated
    default String functionDefinition(Message<?> message) {
        return null;
    }

    default FunctionRoutingResult routingResult(Message<?> message) {
        return new FunctionRoutingResult(this.functionDefinition(message));
    }
    
    Qui c'è un esempio che poi ho utilizzato:
    
    https://docs.spring.io/spring-cloud-function/docs/current/reference/html/spring-cloud-function.html